src/main/java/org/folio/rest/impl/PatronActionSessionAPI.java

```diff
@@ -55,6 +87,45 @@ public void deletePatronActionSessionStoragePatronActionSessionsByPatronSessionI
       DeletePatronActionSessionStoragePatronActionSessionsByPatronSessionIdResponse.class, asyncResultHandler);
   }
 
+  @Override
+  public void getPatronActionSessionStorageExpiredSessionPatronIds(
+    PatronActionSessionStorageExpiredSessionPatronIdsGetActionType actionType, String lastTimeActionLimit,
+    int limit, Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+
+    String tenantId = okapiHeaders.get(RestVerticle.OKAPI_HEADER_TENANT);
+    PgClientFutureAdapter pgClient = PgClientFutureAdapter.create(vertxContext, okapiHeaders);
+    DateTime dateTimeLimit;
+    try {
+      dateTimeLimit = DateTime.parse(lastTimeActionLimit);
+    }
+    catch (Exception e){
+      Errors errors = ValidationHelper.createValidationErrorMessage("last_time_action_limit",
+        lastTimeActionLimit, "cannot be parsed");
+      asyncResultHandler.handle(succeededFuture(
+        PatronActionSessionStorage.PutPatronActionSessionStoragePatronActionSessionsByPatronSessionIdResponse
+          .respond422WithApplicationJson(errors)));
+      return;
+    }
+
+    String sql = toSelectExpiredSessionsQuery(tenantId, ACTION_TYPE_MAP.get(actionType),
+      limit, dateTimeLimit);
+
+    pgClient.select(sql)
+      .map(this::mapPatronIdResponse)
+      .map(GetPatronActionSessionStorageExpiredSessionPatronIdsResponse::respond200WithApplicationJson)
+      .map(Response.class::cast)
+      .otherwise(this::mapExceptionToResponse)
+      .setHandler(asyncResultHandler);
+  }
+
+  private PatronActionExpiredIdsResponse mapPatronIdResponse(ResultSet resultSet) {
+    List<String> patronIds = resultSet.getRows()
+      .stream()
+      .map(json -> json.getString("patron_id"))
```

Comment:
Why is there a JSON property which uses an underscore as a separator?

Reply:
Thank you for review, updated